server/tests/runs.e2e-spec.ts

```diff
@@ -0,0 +1,341 @@
+                        trackNum: 0,
+                        zoneNum: 0,
+                        rank: 111,
+                        createdAt: new Date(),
+                        map: { connect: { id: map2.id } },
+                        user: { connect: { id: user2.id } }
+                    }
+                },
+                baseStats: {
+                    create: {
+                        jumps: 1
+                    }
+                },
+                zoneStats: {
+                    createMany: {
+                        data: [
+                            {zoneNum: 1},
+                            {zoneNum: 2}
+                        ]
+                    }
+                }
+            },
+            include: {
+                baseStats: true,
+                zoneStats: true
+            }
+        });
+
+        const authService: AuthService = global.auth as AuthService;
+        global.accessToken = (await authService.login(user1)).access_token;
+    });
+
+    afterEach(async () => {
+        const prisma: PrismaService = global.prisma;
+
+        await prisma.user.deleteMany({ where: { id: { in: [user1.id, user2.id] } } });
+        await prisma.run.deleteMany({ where: { id: { in: [run1.id, run2.id, run3.id, run4.id] } } });
+        await prisma.map.deleteMany({ where: { id: { in: [map1.id, map2.id] } } });
+    });
+
+    describe('GET /api/v1/runs', () => {
+        const expects = (res) => expect(res.body).toBeValidPagedDto(RunDto);
+        const filter = (x) => {[run1.id, run2.id, run3.id, run4.id].includes(x.id)};
+        it('should respond with a list of runs', async () => {
+            const res = await get('runs', 200);
+            expects(res);
+            expect(res.body.totalCount).toBeGreaterThan(1);
+            expect(res.body.returnCount).toBeGreaterThan(1);
+        });
+
+        it('should respond with a list of runs with take parameter', () => takeTest('runs', expects));
+
+        it('should respond with a list of runs with skip parameter', () => skipTest('runs', expects));
+
+        it('should respond with list of runs filtered by mapID parameter', async () => {
+            const res = await get('runs', 200, { mapID: map1.id });
+            expects(res);
+            expect(res.body.totalCount).toBe(2);
+            expect(res.body.returnCount).toBe(2);
+            expect(res.body.response[0].mapID).toBe(map1.id);
+        });
+
+        it('should respond with a list of runs filtered by userID parameter', async () => {
+            const res = await get('runs', 200, { userID: user1.id });
+            expects(res);
+            expect(res.body.totalCount).toBe(2);
+            expect(res.body.returnCount).toBe(2);
+            expect(res.body.response[0].playerID).toBe(user1.id);
+        });
+
+        it('should respond with a list of runs filtered by a list of run ids', async () => {
+            const ids: String = user1.id + ',' + user2.id;
```

Comment:
Don't need the `String` type here, can be easily inferred from the RHS (also it should be `string`).